### ISSUE TITLE
Confusing error message when Concatenate inserts implicit position-only separator

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2845]
### Discussed in https://github.com/microsoft/pyright/discussions/2844

<div type='discussions-op-text'>

<sup>Originally posted by **cdce8p** January  9, 2022</sup>
With v1.1.206, the example below now triggers an incompatibleOverride error. I guess something that was hidden before #2838. I'm not sure what the best course of action here should be. Just by definition of PEP 612 the error is correct. `self` in `wrapper` is defined as `positional-only`. In practice however it would be unnecessary cumbersome to have to add `/` after any `self`.  I might have mentioned it already elsewhere, but this is also something that's causing us headaches at the mypy side. https://github.com/python/mypy/pull/11847

Any ideas how to best resolve this?

```py
from __future__ import annotations
from typing import Any, Callable, Concatenate, ParamSpec, TypeVar

T = TypeVar("T", bound="Parent")
R = TypeVar("R")
P = ParamSpec("P")


def decorator(
    func: Callable[Concatenate[T, P], R]
) -> Callable[Concatenate[T, P], R]:
    def wrapper(self: T, *args: P.args, **kwargs: P.kwargs) -> R:
        try:
            return func(self, *args, **kwargs)
        except Exception:
            self.error_handling()
            raise
    return wrapper

class Parent:
    def error_handling(self) -> None:
        ...

    def func(self, y: int, **kwargs: Any) -> None:
        ...

class Child(Parent):
    @decorator
    def func(self, y: int, **kwargs: Any) -> None:
        ...

reveal_type(Parent.func)
reveal_type(Child.func)
```

The fallback would probably be to use `cast`. The downside being that the decorator could now be used everywhere since `self` isn't checked anymore.
```py
def decorator(
    func: Callable[P, R]
) -> Callable[P, R]:
    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
        self = cast(Parent, args[0])
        try:
            return func(*args, **kwargs)
        except Exception:
            self.error_handling()
            raise
    return wrapper
```</div>

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.207, which I just published. It will also be included in the next release of pylance.
